src/pfe/file-watcher/server/src/projects/ShellExtensionProject.ts

```diff
@@ -208,26 +235,35 @@ export class ShellExtensionProject implements IExtensionProject {
 
     /**
      * @function
-     * @description Get the build log for the project.
+     * @description Get logs from files or directories.
      *
-     * @param logDirectory <Required | string> - The log location directory.
+     * @param type <Required | String> - The type of log ("build" or "app")
+     * @param logDirectory <Required | String> - The log location directory.
+     * @param projectID <Required | String> - An alphanumeric identifier for a project.
+     * @param containerName <Required | String> - The docker container name.
      *
-     * @returns Promise<BuildLog>
+     * @returns Promise<Array<AppLog | BuildLog>>
      */
-    getBuildLog = async (logDirectory: string): Promise<BuildLog> => {
-        return await logHelper.getBuildLogs(logDirectory, this.config.buildLogs);
+    getLogs = async (type: string, logDirectory: string, projectID: string, containerName: string): Promise<Array<AppLog | BuildLog>> => {
+        if (type.toLowerCase() != "build" && type.toLowerCase() != "app") return;
+        console.log(">>> LOGS ORIGIN: %j", logsOrigin);
+        return await logHelper.getLogs(type, logsOrigin, logDirectory, projectID, containerName);
     }
 
-    /**
-     * @function
-     * @description Get the app log for the project.
-     *
-     * @param logDirectory <Required | String> - The log location directory.
-     *
-     * @returns Promise<AppLog>
-     */
-    getAppLog = async (logDirectory: string): Promise<AppLog> => {
-        return await logHelper.getAppLogs(logDirectory, this.config.appLogs);
+    setLogsOriginFromExtension(): void {
+        if (this.config.buildContainerLogs && (this.config.buildContainerLogs.files || this.config.buildContainerLogs.dirs)) {
```

Comment:
It's better, but if `files` or `dirs` are of the wrong type (for example, it's a number or boolean), would `logHelper` tolerate it?

It may be better to consolidate the sanitization into a private function within the `logHelper` itself

Reply:
I see, tho just to be safe, could we add a runtime type check here:

https://github.com/eclipse/codewind/pull/899/files#diff-44a052830863e32d7b0f6a81d8d83584R264

`const logs = Object.keys(logsOrigin);` 
(make logs an empty array if `logsOrigin` is not an object)

Reply:
Not sure what the runtime check do though? If `logsOrigin` is not of type 
```
interface ILogOriginFolderOrFiles {
    [x: string]: string;
}
```
it will throw an error, even on runtime

Reply:
`typeof logsOrigin === "object"`?

If being the wrong type is causing a runtime error to be thrown, that's the kind of problem we're trying to guard against isn't it?

Reply:
Yup. Good catch. Added that fix:
```
const logs = (typeof logsOrigin === "object" && logsOrigin !== null) ? Object.keys(logsOrigin) : [];
```

Reply:
Actually, improved the condition even better. Thanks to https://stackoverflow.com/a/52478680

```
const logs = (typeof logsOrigin == "object" && logsOrigin instanceof Object && !(logsOrigin instanceof Array)) ? Object.keys(logsOrigin) : [];
```

This will trim it down and only accept `{}` as an object.